pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -456,3 +474,116 @@ func (cloud *CloudProvider) GetInstanceOrError(ctx context.Context, instanceZone
 	glog.V(4).Infof("Got instance %v from zone %v", instanceName, instanceZone)
 	return instance, nil
 }
+
+
+func (cloud *CloudProvider) GetSnapshot(ctx context.Context, snapshotName string) (*compute.Snapshot, error) {
+	svc := cloud.service
+	project := cloud.project
+	glog.V(4).Infof("Getting snapshot %v", snapshotName)
+	snapshot, err := svc.Snapshots.Get(project, snapshotName).Context(ctx).Do()
+	if err != nil {
+		return nil, err
+	}
+	glog.V(4).Infof("Got snapshot %v", snapshotName)
+	return snapshot, nil
+}
+
+func (cloud *CloudProvider) DeleteSnapshot(ctx context.Context, snapshotName string) error {
+	op, err := cloud.service.Snapshots.Delete(cloud.project, snapshotName).Context(ctx).Do()
+	if err != nil {
+		if IsGCEError(err, "notFound") {
+			// Already deleted
+			return nil
+		}
+		return err
+	}
+	err = cloud.waitForGlobalOp(ctx, op)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cloud *CloudProvider) CreateSnapshot(ctx context.Context, volKey *meta.Key, snapshotName string) (*compute.Snapshot, error) {
```

Comment:
Could you also add an E2E test or two to test this new functionality? Feel free to let me know if you want a walkthrough of how the framework works!

Reply:
sure, will add it in a separate PR

Reply:
could you open an issue and link to this pr

Reply:
actually I just added the e2e tests in a second commit. PTAL.